eventuals/terminal.h

```diff
@@ -200,6 +200,20 @@ struct StoppedException : public std::exception {
 
 ////////////////////////////////////////////////////////////////////////
 
+// Using to don't create nested 'std::exception_ptr'.
+template <typename... Args>
+auto make_exception_ptr_or_forward(Args&&... args) {
+  if constexpr (sizeof...(args) > 0) {
+    return std::make_exception_ptr(std::forward<Args>(args)...);
+  }
+}
+
+inline auto make_exception_ptr_or_forward(std::exception_ptr&& error) {
```

Comment:
What happens if somewhere you don't have an rvalue reference? Will it try and call the above case? To make sure that this specialization will always be picked how about we take `error` by value instead.